service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -60,15 +65,44 @@ private ObjectNode toJsonNode(String json) {
     }
   }
 
-  private void collectDatarepoDatasets(
-      AuthenticatedUserRequest user, DatasetsListResponse response) {
+  private List<ObjectNode> collectWorkspaceDatasets(AuthenticatedUserRequest user) {
+    var workspaces = rawlsService.getWorkspaceIdsAndRoles(user);
+    List<Dataset> datasets =
+        datasetDao.find(
+            StorageSystem.TERRA_WORKSPACE,
+            workspaces.stream()
+                .map(workspaceListResponse -> workspaceListResponse.getWorkspace().getWorkspaceId())
+                .toList());
+    List<ObjectNode> workspaceDatasets = new ArrayList<>();
+
+    for (Dataset dataset : datasets) {
```

Comment:
I moved away from streaming for this iteration, just because it seemed more readable this way if we are doing `ObjectNode` objects. Potentially would restart the conversation if we moved back to objects that can be constructed in a more functional way with builders.